roles/vault/files/config/override.hcl

```diff
@@ -1 +1,5 @@
 ui = "{{ vault_ui_enable }}"
+
+service_registration "consul" {
```

Comment:
The documentation seems to indicate that this stanza is optional if we use Consul as the storage backend, which seems to be the case in our [module provider](https://github.com/hashicorp/terraform-aws-vault/tree/v0.13.4)?
> When Consul is configured as the storage backend, Vault implicitly uses Consul for service registration, so the service_registration stanza is not needed

https://www.vaultproject.io/docs/configuration/service-registration#service_registration-stanza

Reply:
Yeah it should be the case, though strangely when I tried it in the derivative project set-up some time back, it wasn't registered.

Guess I could try again and confirm and get back on this.

Reply:
Okay so I have just tested this, it doesn't work without the `service_registration "consul" {`, despite what the docs has suggested. I do agree with you that it was previously working in older `vault` versions though, but just not for this `1.5.4` version.

Either the docs is describing a release version that happens to be different from ours, if not, it could be this particular release has some issue with this.

In any case, I recommend just explicitly stating the `service_registration`, since it will guarantee the service to register to Consul, regardless of the versions / bugs or not.